synthnn/core/accelerated_musical_network.py

```
@@ -232,18 +238,32 @@ def compute_harmonic_state(self, duration: float, sample_rate: float = 44100) ->
+            morph_time = sample_rate
+            sample_rate = 44100
+        elif isinstance(morph_time, str):
+            raise ValueError(
+                "Invalid arguments: if target_mode is string, morph_time must be numeric"
+            )
+
+        if self.mode_detector is not None:
+            current_intervals = self.mode_detector.mode_intervals[self.mode]
+            target_intervals = self.mode_detector.mode_intervals[target_mode]
+        else:
+            warnings.warn(
```

Ruff (0.11.9)

No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)